lectures/olg.md

```diff
@@ -53,866 +67,799 @@ from scipy import optimize
 
-```{code-cell} ipython3
-def simulate_ts(m, x0_values, ts_length):
+for i in range(x):
+    k_grid_next[i] = k_update(k_grid[i], model)
 
-    k_star = optimize.newton(g, 0.2, args=(m,))
-    fig, ax = plt.subplots(figsize=(10, 5))
+fig, ax = plt.subplots(figsize=(6, 6))
 
-    ts = np.zeros(ts_length)
+ymin, ymax = np.min(k_grid_next), np.max(k_grid_next)
 
-    # simulate and plot time series
-    for x_init in x0_values:
-        ts[0] = x_init
-        for t in range(1, ts_length):
-            ts[t] = k_next(ts[t-1], m)
-        ax.plot(np.arange(ts_length), ts, '-o', ms=4, alpha=0.6,
-                label=r'$k_0=%g$' %x_init)
-    ax.plot(np.arange(ts_length), np.full(ts_length, k_star),
-            alpha=0.6, color='red', label=r'$k^*$')
-    ax.legend(fontsize=10)
+ax.plot(k_grid, k_grid_next,  lw=2, alpha=0.6, label='$g$')
+ax.plot(k_grid, k_grid, 'k-', lw=1, alpha=0.7, label='$45^{\circ}$')
 
-    ax.set_xlabel(r'$t$', fontsize=14)
-    ax.set_ylabel(r'$k_t$', fontsize=14)
 
-    plt.show()
-```
+ax.legend(loc='upper left', frameon=False, fontsize=12)
+ax.set_xlabel('$k_t$', fontsize=12)
+ax.set_ylabel('$k_{t+1}$', fontsize=12)
 
-```{code-cell} ipython3
-simulate_ts(m_crra, x0, ts_length)
+plt.show()
 ```
 
-## Exercises
+
+```{solution-end}
+```
 
 
 ```{exercise}
-:label: olg_ex1
+:label: olg_ex2
 
-Solve the equilibrium of the CRRA case numerically using [](equilibrium_crra).
+The 45 degree diagram from the last exercise shows that there is a unique
+positive steady state.
 
-Plot the equilibrium quantity and equilibrium price in the equilibrium plot with the CRRA utility (TODO label and refer the plot generated by the code).
-```
+The positive steady state can be obtained by setting  $k_{t+1} = k_t = k^*$ in [](law_of_motion_capital_crra), which yields
 
+$$
+    k^* = 
+    \frac{(1-\alpha)(k^*)^{\alpha}}
+    {1 + \beta^{-1/\gamma} (\alpha (k^*)^{\alpha-1})^{(\gamma-1)/\gamma}}
+$$
 
-```{solution-start} olg_ex1
-:class: dropdown
-```
+Unlike the log preference case, the CRRA utility steady state $k^*$ 
+cannot be obtained analytically.
 
-To solve the equation we need to turn to Newton's method.
+Instead, solve for $k^*$ using newton's method.
 
-The function `find_Rstar` is used to find $R^*_{t+1}$ by finding
-the zero of equation [](equilibrium_crra) using the helper
-function `find_Rstar_newton` for a given value of $K_t$.
+```
 
-Similary, `find_Kstar` finds the equilibrium quantity $K^*_{t+1}$ using the value of $R^*_{t+1}$.
 
-```{code-cell} ipython3
-def find_Rstar_newton(x, K_prev, model):
-    α, β, γ, L = model.α, model.β, model.u_params['γ'], model.L
-    lhs = L * (1-α) * (K_prev / L)**α
-    lhs /= (1 + β**(-1/γ) * x**((γ-1)/γ))
-    rhs = L * (x / α)**(1/(α-1))
-    return lhs - rhs
+```{solution-start} olg_ex2
+:class: dropdown
 ```
 
-```{code-cell} ipython3
-def find_Rstar(K_prev, model):
-    return optimize.newton(find_Rstar_newton, 0.5, args=(K_prev, model))
+We introduce a function $g$ such that
```

Comment:
Should we consider using other notations than ``$g$`` to distinguish it from the ``$g$`` used in the plot code above?

Reply:
Good idea.  I've changed it to $h$.